contracts/modules/governance/VotePower.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.0;
+
+import "../../interfaces/IGovernance.sol";
+import "../../interfaces/IMemberRoles.sol";
+import "../../interfaces/INXMMaster.sol";
+import "../../interfaces/INXMToken.sol";
+import "../../interfaces/ITokenController.sol";
+
+contract VotePower {
+
+  string constant public name = "NXM balance with delegations";
+  string constant public symbol = "NXMD";
+  uint8 constant public decimals = 18;
+
+  INXMMaster immutable public master;
+
+  enum Role {UnAssigned, AdvisoryBoard, Member}
+
+  constructor(INXMMaster _master) {
+    master = _master;
+  }
+
+  function totalSupply() public view returns (uint) {
+    return INXMToken(master.tokenAddress()).totalSupply();
+  }
+
+  function balanceOf(address member) public view returns (uint) {
+
+    ITokenController tokenController = ITokenController(master.dAppLocker());
+    IMemberRoles memberRoles = IMemberRoles(master.getLatestAddress("MR"));
+    IGovernance governance = IGovernance(master.getLatestAddress("GV"));
+
+    uint delegationId = governance.followerDelegation(member);
+
+    if (delegationId != 0) {
+      (, address leader,) = governance.allDelegation(delegationId);
+      if (leader != address(0)) {
+        return 0;
+      }
+    }
+
+    uint balance = tokenController.totalBalanceOf(member) + 1e18;
+    uint[] memory delegationIds = governance.getFollowers(member);
+
+    for (uint i = 0; i < delegationIds.length; i++) {
+
+      (address follower, address leader,) = governance.allDelegation(delegationIds[i]);
+
+      if (
```

Comment:
there's an extra check in the `_setVoteTally` code:

https://github.com/NexusMutual/smart-contracts/blob/021f721e3e793fa527badf8dd0a297ab5ca74cd2/contracts/modules/governance/Governance.sol#L930

```(now - _lastUpd) > tokenHoldingTime;```

Reply:
Snapshot deals with double voting by default, isn't it? So this isn't needed

Reply:
lgtm